### ISSUE TITLE
Problem with GetAsDataSet function of TResource on DelphiXE2

### ISSUE THREAD

**Issue by jhonatasobrinho** [fabriciocolombo/delphi-rest-client-api#69]
I had a problem trying to use that function on Delphi XE2. I think, someway the compile directive USE_SUPER_OBJECT was defined, but I can't specify where. So, the GetAsDataSet function wasn't avaiable for use.

I solved this problem taking off the {$UNDEF USE_SUPER_OBJECT} directive from {$IFDEF MACOS}, compiling the component again and compiling my project. So it worked. That is a bug or just my fail somewhere?

**Comment by RobertoSchneiders** [comment]
It's not a bug, the GetAsDataset is only available if you are using SuperObject.

Apparently, `USE_SUPER_OBECT` is defined by default ([DelphiRest.inc#L6](https://github.com/fabriciocolombo/delphi-rest-client-api/blob/46c08e2cd5ef831dbb6d6868f68fc65b622d8f2d/src/DelphiRest.inc#L6)), but not for MacOS ([DelphiRest.inc#L203](https://github.com/fabriciocolombo/delphi-rest-client-api/blob/46c08e2cd5ef831dbb6d6868f68fc65b622d8f2d/src/DelphiRest.inc#L203))

I'm not sure why this is defined like that. My guess is some compatibility issues. Maybe @fabriciocolombo know something.

**Comment by jhonatasobrinho** [comment]
Yes. Apparently it's not a bug, but I couldn't access this function. I had to take off the UNDEF USE_SUPER_OBJECT declaration and so, started to work properly.

I don't know why the UNDEF was hit since it was declared on {$IFDEF MACOS} and I'm not in a MacOS by the way. Please @fabriciocolombo, help us! :)

**Comment by RobertoSchneiders** [comment]
so, you don't have the `MACOS` directive defined on your project settings? for sure?

**Comment by jhonatasobrinho** [comment]
Yes, I'm sure!

**Comment by RobertoSchneiders** [comment]
Weird. That seems to be a problem with your environment.

**Comment by fabriciocolombo** [comment]
Are you using the latest version? Before #43, SuperObject was not enabled by default for all environments.

**Comment by jhonatasobrinho** [comment]
I think so. But I'll download it again and try to reach the problem again.
